manager/volume.go

```diff
@@ -712,13 +722,25 @@ func (m *VolumeManager) EngineUpgrade(volumeName, image string) (v *longhorn.Vol
 		}
 	}
 
-	if err := m.CheckEngineImageReadiness(image); err != nil {
+	v, err = m.ds.GetVolume(volumeName)
+	if err != nil {
 		return nil, err
 	}
 
-	v, err = m.ds.GetVolume(volumeName)
+	replicas, err := m.ds.ListVolumeReplicas(v.Name)
```

Comment:
Add a wrapper func `checkEngineImageReadinessForVolume(volume, node string) bool` to the volumeManager and let it do the actual check of replica nodes.

Reply:
Thank you. Fixed